crates/storage-query-datafusion/src/context.rs

```diff
@@ -313,9 +316,27 @@ impl AsRef<SessionContext> for QueryContext {
     }
 }
 
+/// Newtype to add debug implementation.
+#[derive(Clone, derive_more::Debug)]
```

Comment:
It wasn't apparent to me why we need this to be `Debug`? I thought it may be to derive `Debug` in `QueryContext` but that doesn't seem to need it, everything builds fine if I comment it out. Is the newtype more about implementing `SelectPartitions` over `Metadata`?

Reply:
Yes, `SelectPartitions` requires `Debug`. Will update the comment.